File: Java/lab6/src/main/java/org/example/ConsultantPool.java
package org.example;

import java.util.LinkedList;
import java.util.List;

public class ConsultantPool {
    public List<Consultant> free = new LinkedList<Consultant>() {{
        add(new Consultant("Ivanius", "Popodolopus", 20));
        add(new Consultant("Okun", "Karas", 20));
        add(new Consultant("Karas", "Karas", 20));
    }};
    public List<Consultant> used = new LinkedList<>();
    public Consultant getConsultant() {
        if (free.isEmpty()) {
            return null;
        } else {
            Consultant consultant = free.get(0);
            used.add(consultant);
            free.remove(consultant);
            return consultant;
        }
    }
    public void returnConsultant(Consultant consultant) {
        used.remove(consultant);
        free.add(consultant);
    }
}

File: Python/lab6/ConsultantPool.py
from Consultant import Consultant

class ConsultantPool:
    free = [Consultant("Roma", "Ppavliuk", 20), Consultant("Okun", "Karas", 20), Consultant("Karas", "Karas", 20)]
    used = []
    def getConsultant(self) -> Consultant:
        if not self.free:
            return None
        else:
            consultant: Consultant = self.free.pop(0)
            self.used.append(consultant)
            return consultant
    def returnConsultant(self, consultant: Consultant):
        self.used.remove(consultant)
        self.free.append(consultant)